### PR TITLE
fix: [PL-23715]: Fix npe in to yaml for wf

### PR DESCRIPTION
**Pull request by abhinav-harness**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/31411)

### CODE REVIEW

**Comment by abhinav-harness** [comment]
trigger messagecheck

File: 400-rest/src/main/java/software/wings/service/impl/workflow/creation/K8V2RollingWorkflowCreator.java [hunk 1]
```diff
@@ -50,10 +50,11 @@ public Workflow createWorkflow(Workflow clientWorkflow) {
...
     notNullCheck("orchestrationWorkflow", canaryOrchestrationWorkflow);
-    K8AbstractWorkflowHelper k8AbstractWorkflowHelper =
-        infrastructureDefinition.getInfrastructure().getInfrastructureType().equals("RANCHER_KUBERNETES")
-        ? rancherK8RollingWorkflowPhaseHelper
-        : k8RollingWorkflowPhaseHelper;
+    K8AbstractWorkflowHelper k8AbstractWorkflowHelper = k8RollingWorkflowPhaseHelper;
+    if (infrastructureDefinition != null
+        && infrastructureDefinition.getInfrastructure().getInfrastructureType().equals("RANCHER_KUBERNETES")) {
+      k8AbstractWorkflowHelper = rancherK8RollingWorkflowPhaseHelper;
+    }
```
Comment (shubham-in10se) [inline 400-rest/src/main/java/software/wings/service/impl/workflow/creation/K8V2RollingWorkflowCreator.java:57]: check this for Canary and Blue-Green workflow creators, we are doing something similar there as well.

**Review by shubham-in10se — APPROVED** [review]
validate the same for canary and BG workflow creators as well.

**Review by meenakshi-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31411)* status: :shipit: complete! ✓

**Pull request merged**